### PR TITLE
CL: Add local auth mount entity feature

### PR DESCRIPTION
**Pull request by vishalnayak**

### CODE REVIEW

**Review by HridoyRoy — COMMENTED** [review]
I think changelog entries for enterprise features are made to the ent repo now.

**Review by HridoyRoy — APPROVED** [review]
Ah, sorry -- this is the changelog entry for the full set of PRs for the local auth mounts project, my mistake. Looks good to me!

**Review by mladlow — CHANGES_REQUESTED** [review]
If this merges to main right now it will show up in the 1.10 changelog. Can you make this PR directly against the 1.9 release branch please? 

Also, if the changelog should show up as a feature, could you please review the formatting guidelines on the wiki, and either update the formatting or make this note an improvement?

File: changelog/12848.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+```release-note:feature
```
Comment (mladlow) [inline changelog/12848.txt:1]: This is a very informative new feature writeup on this change!

Comment (vishalnayak) [reply changelog/12848.txt:1]: Thanks!

**Comment by vishalnayak** [comment]
@mladlow Closing in favor of https://github.com/hashicorp/vault/pull/13113

**Pull request closed without merging**